src/test/unit/DelegationUnit.t.sol

```diff
@@ -47,275 +40,459 @@ contract DelegationUnitTests is EigenLayerTestHelper {
+     */
+    function _getApproverSignature(
+        uint256 _delegationSignerPrivateKey,
+        address staker,
+        address operator,
+        bytes32 salt,
+        uint256 expiry
+    ) internal view returns (ISignatureUtils.SignatureWithExpiry memory approverSignatureAndExpiry) {
+        approverSignatureAndExpiry.expiry = expiry;
+        {
+            bytes32 digestHash = delegationManager.calculateDelegationApprovalDigestHash(
+                staker,
+                operator,
+                delegationManager.delegationApprover(operator),
+                salt,
+                expiry
+            );
+            (uint8 v, bytes32 r, bytes32 s) = cheats.sign(_delegationSignerPrivateKey, digestHash);
+            approverSignatureAndExpiry.signature = abi.encodePacked(r, s, v);
+        }
+        return approverSignatureAndExpiry;
     }
 
     /**
-     * @notice `operator` registers via calling `DelegationManager.registerAsOperator(operatorDetails, metadataURI)`
-     * Should be able to set any parameters, other than setting their `earningsReceiver` to the zero address or too high value for `stakerOptOutWindowBlocks`
-     * The set parameters should match the desired parameters (correct storage update)
-     * Operator becomes delegated to themselves
-     * Properly emits events – especially the `OperatorRegistered` event, but also `StakerDelegated` & `OperatorDetailsModified` events
-     * Reverts appropriately if operator was already delegated to someone (including themselves, i.e. they were already an operator)
-     * @param operator and @param operatorDetails are fuzzed inputs
+     * @notice internal function for calculating a signature from the staker corresponding to `_stakerPrivateKey`, delegating them to
+     * the `operator`, and expiring at `expiry`.
      */
-    function testRegisterAsOperator(address operator, IDelegationManager.OperatorDetails memory operatorDetails, string memory metadataURI) public 
-        filterFuzzedAddressInputs(operator)
-    {
+    function _getStakerSignature(
+        uint256 _stakerPrivateKey,
+        address operator,
+        uint256 expiry
+    ) internal view returns (ISignatureUtils.SignatureWithExpiry memory stakerSignatureAndExpiry) {
+        address staker = cheats.addr(stakerPrivateKey);
+        stakerSignatureAndExpiry.expiry = expiry;
+        {
+            bytes32 digestHash = delegationManager.calculateCurrentStakerDelegationDigestHash(staker, operator, expiry);
+            (uint8 v, bytes32 r, bytes32 s) = cheats.sign(_stakerPrivateKey, digestHash);
+            stakerSignatureAndExpiry.signature = abi.encodePacked(r, s, v);
+        }
+        return stakerSignatureAndExpiry;
+    }
+
+    // @notice Assumes operator does not have a delegation approver & staker != approver
+    function _delegateToOperatorWhoAcceptsAllStakers(address staker, address operator) internal {
+        cheats.assume(staker != operator);
```

Comment:
I think putting 'assumes' inside of internal helpers  like this is a bit dangerous, in general.
Here it makes sense, just noting that we should be careful about when we do this, because it makes the set of (fuzzed) inputs that we're filtering out a bit more opaque.

Reply:
Good point, moved the assumes out of the internals and added to tests as needed https://github.com/Layr-Labs/eigenlayer-contracts/pull/332/commits/d9645a9e4a9d32a3526cb5827b8ef30f3d85c5c4